ta/gatekeeper/ta_gatekeeper.h

```diff
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright (c) 2018, Linaro Limited */
+/* Copyright (c) 2017, GlobalLogic  */
+
+#ifndef __TA_GATEKEEPER_H
+#define __TA_GATEKEEPER_H
+
+#include <compiler.h>
+#include <tee_internal_api.h>
+#include <stdint.h>
+
+/*
+ * Please keep password_handle_t structure consistent with its counterpart
+ * which defined in system/gatekeeper/include/gatekeeper/password_handle.h
+ */
+
+#define HANDLE_VERSION			2
+#define HANDLE_VERSION_THROTTLE		2
+#define HANDLE_FLAG_THROTTLE_SECURE	1
+
+#define HW_AUTH_TOKEN_VERSION		0
+
+#define KM_GET_AUTHTOKEN_KEY		65536
+
+/*
+ * Please keep this variable consistent with TA_UUID variable that
+ * is defined in Keymaster Android.mk file
+ */
+#define TA_KEYMASTER_UUID { 0xdba51a17, 0x0563, 0x11e7, \
+			{ 0x93, 0xb1, 0x6f, 0xa7, 0xb0, 0x07, 0x1a, 0x51} }
+
+/*
+ * Please keep this define consistent with KM_GET_AUTHTOKEN_KEY constant that
+ * is defined in Keymaster
+ */
+
+typedef uint64_t secure_id_t;
+typedef uint64_t salt_t;
+
+struct __packed password_handle {
+	uint8_t version;
+	secure_id_t user_id;
+	uint64_t flags;
+	salt_t salt;
+	uint8_t signature[32];
+	bool hardware_backed;
+};
+
+/*
+ * Please keep hw_auth_token_t structure consistent with its counterpart
+ * which defined in hardware/libhardware/include/hardware/hw_auth_token.h
+ */
+enum hw_authenticator {
+	HW_AUTH_NONE = 0,
+	HW_AUTH_PASSWORD = 1 << 0,
+	HW_AUTH_FINGERPRINT = 1 << 1,
+	/* Additional entries should be powers of 2. */
+	HW_AUTH_ANY = (int)((uint32_t) ~0U)
```

Comment:
That's a fancy way of writing a bunch of "F's".

Reply:
fixed